keyboards/mawaeg/sharkropad/sharkropad.c

```diff
@@ -0,0 +1,126 @@
+/* Copyright 2025 @mawaeg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ #include "quantum.h"
+ #include <ctype.h>
+ #include <rgb_matrix.h>
+ 
+ #if defined(RGB_MATRIX_EFFECT)
+ #    undef RGB_MATRIX_EFFECT
+ #endif // defined(RGB_MATRIX_EFFECT)
+ 
+ #define RGB_MATRIX_EFFECT(x) RGB_MATRIX_EFFECT_##x,
+ enum {
+     RGB_MATRIX_EFFECT_NONE,
+ #include "rgb_matrix_effects.inc"
+ #ifdef RGB_MATRIX_CUSTOM_KB
+ #    include "rgb_matrix_kb.inc"
+ #endif // RGB_MATRIX_CUSTOM_KB
+ #ifdef RGB_MATRIX_CUSTOM_USER
+ #    include "rgb_matrix_user.inc"
+ #endif // RGB_MATRIX_CUSTOM_USER
+ #undef RGB_MATRIX_EFFECT
+ };
+ 
+ #define RGB_MATRIX_EFFECT(x)    \
+     case RGB_MATRIX_EFFECT_##x: \
+         return #x;
+ const char *rgb_matrix_name(uint8_t effect) {
+     switch (effect) {
+         case RGB_MATRIX_EFFECT_NONE:
+             return "NONE";
+ #include "rgb_matrix_effects.inc"
+ #ifdef RGB_MATRIX_CUSTOM_KB
+ #    include "rgb_matrix_kb.inc"
+ #endif // RGB_MATRIX_CUSTOM_KB
+ #ifdef RGB_MATRIX_CUSTOM_USER
+ #    include "rgb_matrix_user.inc"
+ #endif // RGB_MATRIX_CUSTOM_USER
+ #undef RGB_MATRIX_EFFECT
+         default:
+             return "UNKNOWN";
+     }
+ }
+ 
+ static uint8_t effect_name_len = 0;
+ 
+ const char *rgb_matrix_get_effect_name(void) {
+     static char buf[32] = {0};
+ 
+     snprintf(buf, sizeof(buf), "%s", rgb_matrix_name(rgb_matrix_get_mode()));
+     for (uint8_t i = 1; i < sizeof(buf); ++i) {
+         if (buf[i] == 0) {
+             effect_name_len = i;
+             break;
+         } else if (buf[i] == '_')
+             buf[i] = ' ';
+         else if (buf[i - 1] == ' ')
+             buf[i] = toupper(buf[i]);
+         else if (buf[i - 1] != ' ')
+             buf[i] = tolower(buf[i]);
+     }
+     return buf;
+ }
+ 
+bool oled_task_kb(void) {
+    if (!oled_task_user()) { return false; }
+
+     static uint8_t  last_effect = 0;
+     static uint8_t  last_speed  = 0;
+     static uint16_t key_timer   = 0;
+     static uint8_t  start_index = 0;
+ 
+     uint8_t speedPercentage  = (uint8_t)(((float)rgb_matrix_get_speed() / 255.0) * 100.0);
```

Comment:
Just a heads up, we do use lib8tion for rgb matrix, and as such, you should have access to those functions.   I mention that, because these are fast/efficient 8 bit math functions. 

```suggestion
    uint8_t speedPercentage = scale8(rgb_matrix_get_speed(), 100);
```

I do realize that the STM32L432 does have a floating point unit, but generally, avoiding floats is better, if you can avoid it. Especially if somebody decides to copy the code for an AVR board or something that doesn't have a FPU.

Reply:
I applied those changes in b1c22b2 as the include for lib8tion is also needed.